src/sync/impl/sync_client.hpp

```diff
@@ -26,6 +26,10 @@
 
 #include <thread>
 
+#if REALM_PLATFORM_APPLE && !TARGET_OS_WATCH
```

Comment:
I don't like repeating the limitation that all Apple platforms other than watchOS support reachability in multiple places like this. It'd be preferable to include the header on all Apple platforms and have it expose a macro that we can use to determine whether reachability is available, and then use that macro to guard use of the reachability class.